rts/lib/CMakeLists.txt

```diff
@@ -85,3 +85,25 @@ option(TRACY_ON_DEMAND "Enable tracy profiling" ON)
 option(TRACY_PROFILE_MEMORY "Profile memory allocations" OFF)
 
 add_subdirectory(tracy)
+
+if (TRACY_ENABLE)
+	option(RMLUI_TRACY_PROFILING "Enable RmlUi profiling with Tracy." ON)
+	# We have our own memory profiling, never enable RmlUi one.
+	set(RMLUI_TRACY_MEMORY_PROFILING OFF CACHE BOOL "Enable RmlUi tracy profiling")
+endif()
+
+# We resolve freetype here to make sure RmlUi find_package picks the same
+# correct version as picked by the engine.
+# !!! If you change freetype resolution here, change it also in rts/builds/legacy
+find_freetype_hack() # hack to find different named freetype.dll
+find_package_static(Freetype REQUIRED)
```

Comment:
I might be experiencing an issue with Freetype on Linux, but I might also just have messed up my build or setup.

The crash I get is:

```
zenity: symbol lookup error: /lib/x86_64-linux-gnu/libcairo.so.2: undefined symbol: FT_Get_Transform
Error: Spring crashed
```

Can anyone else confirm or deny that they get the same issue on Linux?

I built and ran the engine on the main branch without issue.

I ran the different engines using the debug launcher.

Reply:
@p2004a Do you have any ideas or suggestions on this topic? If the inclusion of Freetype was hacky before the RmlUi integration, then I could suspect that it could be tricky to fix Freetype dependency handling, if that is the issue. But it could also just be my build or setup that is at fault.

Reply:
I do not understand the purpose of the Freetype hack that was made before these changes and that are present in the main branch. I found this change from 10 years ago that introduced the hack: https://github.com/beyond-all-reason/spring/commit/446c5a5e9bc04eeb255f57758deecb8f8fda30d6 . I do not understand the purpose of that code either. Before that, there was a Freetype-specific CMake file, similar to the CMake files in https://github.com/beyond-all-reason/spring/tree/BAR105/rts/build/cmake . Is the purpose to help link to libraries/external dependencies present in the environment, and also handle different platforms, and environments like MinGW? I wonder if greenfield projects would prefer to use package management tools like conan.io or vcpkg, though there can be advantages to not having to build external dependencies and instead use a system's available libraries and packages.

Reply:
Please disregard, removing the whole build folder and building again appear to have fixed things.